doc/esmvalcore/preprocessor.rst

```diff
@@ -3,41 +3,75 @@
-* `Land/Sea/Ice masking`_.
-* `Horizontal regridding`_.
-* `Masking of missing values`_.
-* `Multi-model statistics`_.
-* `Time operations`_.
-* `Area operations`_.
-* `Volume operations`_.
-* `Unit conversion`_.
+* :ref:`Variable derivation`
+* :ref:`CMOR check and dataset-specific fixes`
+* :ref:`Vertical interpolation`
+* :ref:`Land/Sea/Ice masking`
+* :ref:`Horizontal regridding`
+* :ref:`Masking of missing values`
+* :ref:`Multi-model statistics`
+* :ref:`Time operations`
+* :ref:`Area operations`
+* :ref:`Volume operations`
+* :ref:`Unit conversion`
+
+Overview
+========
+
+.. 
+   ESMValTool is a modular ``Python 3.6+`` software package possesing capabilities
+   of executing a large number of diagnostic routines that can be written in a
+   number of programming languages (Python, NCL, R, Julia). The modular nature
+   benefits the users and developers in different key areas: a new feature
+   developed specifically for version 2.0 is the preprocessing core  or the
+   preprocessor (esmvalcore) that executes the bulk of standardized data
+   operations and is highly optimized for maximum performance in data-intensive
+   tasks. The main objective of the preprocessor is to integrate as many
+   standardizable data analysis functions as possible so that the diagnostics can
+   focus on the specific scientific tasks they carry. The preprocessor is linked
+   to the diagnostics library and the diagnostic execution is seamlessly performed
+   after the preprocessor has completed the its steps. The benefit of having a
+   preprocessing unit separate from the diagnostics library include:
+
+   * ease of integration of new preprocessing routines;
+   * ease of maintenance (including unit and integration testing) of existing
+     routines;
+   * a straightforward manner of importing and using the preprocessing routines as
+     part  of the overall usage of the software and, as a special case, the use
+     during diagnostic execution;
+   * shifting the effort for the scientific diagnostic developer from implementing
+     both standard and diagnostic-specific functionalities to allowing them to
+     dedicate most of the effort to developing scientifically-relevant diagnostics
+     and metrics;
+   * a more strict code review process, given the smaller code base than for
+     diagnostics.
 
+The ESMValTool preprocessor can be used to perform a broad range of operations
+on the input data before diagnostics or metrics are applied. The preprocessor
+performs these operations in a centralized, documented and efficient way, thus
+reducing the data processing load on the diagnostics side.
+
+Each of the preprocessor operations is written in a dedicated python module and
+all of them receive and return an Iris `cube
+<https://scitools.org.uk/iris/docs/v2.0/iris/iris/cube.html>`_ , working
+sequentially on the data with no interactions between them. The order in which
+the preprocessor operations is applied is set by default in order to minimize
```

Comment:
Is this "in order" necessary here? It makes the sentence slightly confusing.

Reply:
Done.